crates/ruma-client-api/src/unversioned/get_supported_versions.rs

```diff
@@ -45,15 +45,21 @@ impl Response {
     /// Extracts known Matrix versions from this response.
     ///
     /// Matrix versions that Ruma cannot parse, or does not know about, are discarded.
-    pub fn known_versions(&self) -> Vec<MatrixVersion> {
-        let mut versions = vec![];
-        for s in &self.versions {
-            if let Ok(ver) = s.as_str().try_into() {
-                if !versions.contains(&ver) {
-                    versions.push(ver)
-                }
-            }
-        }
-        versions
+    ///
+    /// The versions returned will be sorted from oldest to latest. Use [`.find()`][Iterator::find]
+    /// or [`.rfind()`][DoubleEndedIterator::rfind] to look for a minimum or maximum version to use
+    /// given some constraint.
+    pub fn known_versions(&self) -> impl Iterator<Item = MatrixVersion> + DoubleEndedIterator {
+        self.versions
+            .iter()
+            // Parse, discard unknown versions
+            .flat_map(|s| s.parse::<MatrixVersion>())
+            // Map to key-value pairs where the key is the major-minor representation
```

Comment:
Maybe also place a comment that this sorts it implicitly together with `collect::<BTreeMap>`, maybe also have a test for it.